consul/consul.go

```diff
@@ -70,6 +79,14 @@ func (c Client) Get(k string, v interface{}) (found bool, err error) {
 	return true, c.codec.Unmarshal(data, v)
 }
 
+func (c Client) get(k string) (*api.KVPair, error) {
+	c.m.Lock()
```

Comment:
If the `api.KV` really needs manual locking, shouldn't locking be enough for write operations?

If not, then maybe a `RWMutex` could be better than a `Mutex`? It would allow multiple concurrent read operations, but only a single write operation at a time, and also no write operation while a read operation is ongoing.
I think it could lead to higher performance for read-heavy workloads and doesn't have many downsides for write-operations, right?